src/common/checksums.cpp

```diff
@@ -142,20 +142,21 @@ QByteArray makeChecksumHeader(const QByteArray &checksumType, const QByteArray &
     return header;
 }
 
-QByteArray findBestChecksum(const QByteArray &checksums)
+QByteArray findBestChecksum(const QByteArray &_checksums)
 {
+    const auto checksums = QString::fromUtf8(_checksums);
     int i = 0;
     // The order of the searches here defines the preference ordering.
-    if (-1 != (i = checksums.indexOf("SHA3-256:"))
-        || -1 != (i = checksums.indexOf("SHA256:"))
-        || -1 != (i = checksums.indexOf("SHA1:"))
-        || -1 != (i = checksums.indexOf("MD5:"))
-        || -1 != (i = checksums.indexOf("Adler32:"))) {
+    if (-1 != (i = checksums.indexOf(QLatin1String("SHA3-256:"), 0, Qt::CaseInsensitive))
+        || -1 != (i = checksums.indexOf(QLatin1String("SHA256:"), 0, Qt::CaseInsensitive))
+        || -1 != (i = checksums.indexOf(QLatin1String("SHA1:"), 0, Qt::CaseInsensitive))
+        || -1 != (i = checksums.indexOf(QLatin1String("MD5:"), 0, Qt::CaseInsensitive))
+        || -1 != (i = checksums.indexOf(QLatin1String("ADLER32:"), 0, Qt::CaseInsensitive))) {
         // Now i is the start of the best checksum
         // Grab it until the next space or end of string.
-        auto checksum = checksums.mid(i);
-        return checksum.mid(0, checksum.indexOf(" "));
+        return _checksums.mid(i, _checksums.indexOf(' ', i) - i);
```

Comment:
Quick comment here since I've been stumbling over this code today. In case the client is using this method to also handle parsing of the oc:checksums dav property it might make sense to also pull in the change from https://github.com/owncloud/client/commit/efe18f3afafe392256988ac28958a5f259c7083d

The propfind response would contain something like this, which would to my understanding of the code include the closing tag in the returned checksum:

```
<oc:checksums><oc:checksum>SHA1:fcf5e0cfee14b3e728034956e078b2786db82a07</oc:checksum></oc:checksums>
```

would return:

```
SHA1:fcf5e0cfee14b3e728034956e078b2786db82a07</oc:checksum>
```

The patch mentioned would also help us to align the dav response to a more sane format for the server response in the future as it could also handle individual values:

```
<oc:checksums>
  <oc:checksum>SHA1:asdasd</oc:checksum>
  <oc:checksum>SHA2:sidugsidgu</oc:checksum>
  <oc:checksum>MD5:34234</oc:checksum>
</oc:checksums>
```